parser.y

```diff
@@ -1275,6 +1275,15 @@ AlterTableSpec:
 			PartitionNames: $3.([]model.CIStr),
 		}
 	}
+|	"TRUNCATE" "PARTITION" "ALL"
```

Comment:
how about this:
```
PartitionNameListOrAll:
ALL
| PartitionNameList
```

Reply:
and could you fix the following case by the way?
```
  | DISCARD PARTITION {partition_names | ALL} TABLESPACE
  | IMPORT PARTITION {partition_names | ALL} TABLESPACE
  | TRUNCATE PARTITION {partition_names | ALL}
  | ANALYZE PARTITION {partition_names | ALL}
  | CHECK PARTITION {partition_names | ALL}
  | OPTIMIZE PARTITION {partition_names | ALL}
  | REBUILD PARTITION {partition_names | ALL}
  | REPAIR PARTITION {partition_names | ALL}
```

Reply:
Thanks for your suggestions, and I'm willing to fix all these cases. 😄

Reply:
```
  | CHECK PARTITION {partition_names | ALL}
  | REBUILD PARTITION {partition_names | ALL}
```
the CHECK and REBUILD were claimed by @spongedu

Reply:
OK